### PR TITLE
feat(history-plugin): allow custom clear function

### PR DESCRIPTION
**Pull request by flmariano**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/datorama/akita/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: [227](https://github.com/datorama/akita/issues/227)

## What is the new behavior?

The state history can now be cleared in custom ways, e.g. only the future.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: akita/src/plugins/stateHistory/stateHistoryPlugin.ts [hunk 1]
```diff
@@ -3,12 +3,19 @@ import { toBoolean } from '../../toBoolean';
...
+import { isFunction } from '../../isFunction';
 
 export interface StateHistoryParams {
   maxAge?: number;
   comparator?: (prevState, currentState) => boolean;
 }
 
+export type History<S> = {
+  past: S[],
+  present: S,
```
Comment (NetanelBasal) [inline akita/src/plugins/stateHistory/stateHistoryPlugin.ts]: present should be `S | null`

Comment (flmariano) [reply akita/src/plugins/stateHistory/stateHistoryPlugin.ts]: Ok. What should the commit message for an adjustment like that look like?

**Comment by NetanelBasal** [comment]
use the `amend` command. https://medium.com/@igor_marques/git-basics-adding-more-changes-to-your-last-commit-1629344cb9a8

File: akita/src/plugins/stateHistory/stateHistoryPlugin.ts (hunk 1 above)
Comment (NetanelBasal) [reply akita/src/plugins/stateHistory/stateHistoryPlugin.ts]: use the amend command. https://medium.com/@igor_marques/git-basics-adding-more-changes-to-your-last-commit-1629344cb9a8

**Pull request merged**